sycl/test/basic_tests/buffer/buffer.cpp

```diff
@@ -1,3 +1,5 @@
+// REQUIRES: opencl
```

Comment:
As there is no OpenCL used in this test it should not `REQUIRES: opencl`. Is there a specific reason why you added it?

Reply:
@rbegam I am holding off approval because I might not learn why this is needed and would be tempted to remove it in the PR I am preparing and then break something unknowingly. Other than this reason the PR looks great!

Reply:
@bjoernknafla, take a look at line 568 of modified file.
```c++
program prog(myQueue.get_context());
prog.build_with_source("kernel void override_source(global int* Acc) "
                          "{Acc[get_global_id(0)] = 0; }\n");
cl::sycl::kernel krn = prog.get_kernel("override_source");
```
I think this can be executed only through OpenCL or at least some API, which supports compilation of OpenCL C programs.

Reply:
Oh, I only searched for `cl_` and therefore missed this. Thank you and sorry for not seeing this @rbegam !

Reply:
sorry for my late response. thanks to @bader for attending the query.